internal/builder/android.go

```diff
@@ -30,7 +28,7 @@ func (b *Android) PreBuild(vol *volume.Volume, opts PreBuildOptions) error {
 	if opts.AppID == "" {
 		return fmt.Errorf("appID is required for android build")
 	}
-	return goModInit(vol, opts.Verbose)
+	return goModInit(b, vol, opts.Verbose)
```

Comment:
```suggestion
	return goModInit(b.DockerImage(), vol, opts.Verbose)
```
What about passing only the docker image now that is defined the DockerImage method into the interface.

Reply:
Note that `DockerImage()` can return an error too. Passing the image name would require an error check at every call site. Passing the Builder seems like a cleaner design to me, since `goModInit()` won't even use the `Builder` nor the image name itself, it just passes it to `runBuilderDockerCmd()` which serves as an entry point of all docker related commands run by a builder. Also, at some point in the future you might want to check some attributes of the Builder in either `goModInit()` or `runBuilderDockerCmd()`. By passing the Builder you don't need to change the function signature.

Reply:
I was referring to the `DockerImage() string` method defined into the Builder interface that returns only a string.

Please don't get me wrong, I fully agree with you that passing an interface is a cleaner design.
What I'm saying is that in this case considering that since the code in not exposed (internal package) and that the method returns a string it could be good enough to left the `goModInit()` and `dockerCmd()` signatures as is accepting a string.

If we want to go anyway for a clean desing we'd also define for something like:

```
type DockerImager interface {
   DockerImage() string
}
```
embed the `DockerImager` into the `Builder` interface and then use the `DockerImager` into the `goModInit()` and `dockerCmd()` signatures. WDYT?